src/Scoreboard.js

```diff
@@ -7,35 +7,94 @@ export function update(game) {
 }
 
 export function updateTopScore(game, firebase) {
-  updateUserTopScore(game);
-  updateGlobalTopScore(game, firebase);
+  updateLocalTopScore(game);
+  checkForNewGlobalTopScore(game, firebase);
 }
 
-function updateUserTopScore(game) {
+function updateLocalTopScore(game) {
   if(game.state.currentScore > game.state.topScore){
     game.setState({topScore: game.state.currentScore});
     localStorage.topscore = game.state.currentScore;
   }
 }
 
-function updateGlobalTopScore(game, firebase) {
-  globalTopScore(firebase, game);
-  if(game.state.currentScore > game.state.globalTopScore){
-    firebase.set({
-      topScore: game.state.topScore
-    });
-    game.setState({globalTopScore: game.state.currentScore});
+function checkForNewGlobalTopScore(game, firebase) {
+  globalScoreboard(firebase, game);
+  if(game.state.currentScore > game.state.globalScoreBoard.topScores.fifth.score){
+    game.setState({newTopScore: true});
+  }
+}
+
+function setNewScoreAndUser(scoreboard, name, currentScore, rank) {
+  scoreboard[rank].score = currentScore
+  scoreboard[rank].name = name
+}
+
+function updateRank(scoreboard, newRank, oldRank) {
+  scoreboard[newRank].score = scoreboard[oldRank].score
+  scoreboard[newRank].name = scoreboard[oldRank].name
+}
+
+function updateFirst(scoreboard, name, currentScore) {
+  updateRank(scoreboard, 'fifth', 'fourth')
+  updateRank(scoreboard, 'fourth', 'third')
+  updateRank(scoreboard, 'third', 'second')
+  updateRank(scoreboard, 'second', 'first')
+  setNewScoreAndUser(scoreboard, name, currentScore, 'first')
+}
+
+function updateSecond(scoreboard, name, currentScore) {
+  updateRank(scoreboard, 'fifth', 'fourth')
+  updateRank(scoreboard, 'fourth', 'third')
+  updateRank(scoreboard, 'third', 'second')
+  setNewScoreAndUser(scoreboard, name, currentScore, 'second')
+}
+
+function updateThird(scoreboard, name, currentScore) {
+  updateRank(scoreboard, 'fifth', 'fourth')
+  updateRank(scoreboard, 'fourth', 'third')
+  setNewScoreAndUser(scoreboard, name, currentScore, 'third')
+}
+
+function updateFourth(scoreboard, name, currentScore) {
+  updateRank(scoreboard, 'fifth', 'fourth')
+  setNewScoreAndUser(scoreboard, name, currentScore, 'fourth')
+}
+
```

Comment:
yeah we def should refactor @weilandia. I think that it could def be an array or something and we just input wherever in the rank. you should've seen what I originally started with ;) haha